### PR TITLE
feature/send_text_with_geo

### PR DESCRIPTION
**Pull request by teamagam1**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/76" target="_blank">ReviewNinja

File: GimelGimel.iml [hunk 1]
```diff
@@ -13,7 +13,7 @@
     <content url="file://$MODULE_DIR$">
       <excludeFolder url="file://$MODULE_DIR$/.gradle" />
     </content>
-    <orderEntry type="jdk" jdkName="1.7" jdkType="JavaSDK" />
+    <orderEntry type="jdk" jdkName="1.8" jdkType="JavaSDK" />
```
Comment (teamagam3) [inline GimelGimel.iml]: Undo changes, should be 1.7 as we speaked !resolve

File: app/src/debug/java/com/teamagam/gimelgimel/app/utils/Constants.java [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package com.teamagam.gimelgimel.app.utils;
 
 public class Constants {
-    public static final String MESSAGING_SERVER_URL = "http://ggmessaging-dev2.herokuapp.com";
```
Comment (teamagam3) [inline app/src/debug/java/com/teamagam/gimelgimel/app/utils/Constants.java:4]: Do we need to use this server?
Revert if no

File: app/src/main/java/com/teamagam/gimelgimel/app/model/ViewsModels/MessageMapEntitiesViewModel.java [hunk 3]
```diff
@@ -29,32 +32,29 @@ public void onEntityClick(Entity entity) {
...
     private PointGeometry getPointGeometry(Message message) {
         switch (message.getType()) {
-            case Message.LAT_LONG:
-                return ((MessageLatLong) message).getContent();
+            case Message.GEO:
+                return ((MessageGeo) message).getContent().getPointGeometry();
             case Message.IMAGE:
                 return ((MessageImage) message).getContent().getLocation();
             default:
-                throw new IllegalArgumentException("Message added to map should be supported type.");
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/model/ViewsModels/MessageMapEntitiesViewModel.java:46]: Do we really want to throw an exception on unsupported type?
I like to return null instead.

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java [hunk 4]
```diff
@@ -386,58 +382,13 @@ private void addPinPoint(PointGeometry pointGeometry, VectorLayer vectorLayer) {
...
-            return new PointTextSymbol(Constants.ACTIVE_USER_LOCATION_PIN_CSS_COLOR,
-                    userLocation.getId(), Constants.USER_LOCATION_PIN_SIZE_PX);
-        }
-
-        private Symbol createStaleUserLocationSymbol(UserLocation userLocation) {
-            return new PointTextSymbol(Constants.STALE_USER_LOCATION_PIN_CSS_COLOR,
-                    userLocation.getId(),
-                    Constants.USER_LOCATION_PIN_SIZE_PX);
-        }
+    public interface MessageSymbolizer {
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java]: Why does the ViewFragment have interface in it?
Extract to new file.

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendGeographicMessageDialog.java [hunk 5]
```diff
@@ -75,12 +87,31 @@ public void onCreate(Bundle savedInstanceState) {
...
     @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putParcelable(ARG_POINT_GEOMETRY, mPoint);
+    }
+
+    private void initSpinner() {
+
+        // Create an ArrayAdapter using the string array and a default spinner layout
+        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(getActivity(),
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendGeographicMessageDialog.java]: Convert CharSequence to String, should be the same thing and you shouldn't have to do conversion between them.
Convert if no "special" conversion needed.

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendGeographicMessageDialog.java [hunk 6]
```diff
@@ -75,12 +87,31 @@ public void onCreate(Bundle savedInstanceState) {
     }
 
     @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putParcelable(ARG_POINT_GEOMETRY, mPoint);
+    }
+
+    private void initSpinner() {
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendGeographicMessageDialog.java]: Arrange methods, public methods before privates.
Private is the last in the sequence.

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendGeographicMessageDialog.java [hunk 7]
```diff
@@ -125,18 +157,32 @@ protected boolean hasPositiveButton() {
     protected void onPositiveClick() {
         sLogger.userInteraction("Clicked OK");
 
-        mMessageSender.sendLatLongMessageAsync(mPoint);
+        if (!validateDialogInput()) {
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendGeographicMessageDialog.java]: To make it more clear, put the logic lines inside the if statement and invert it.

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/base/BaseDialogFragment.java [hunk 8]
```diff
@@ -140,6 +143,7 @@ public void onClick(android.content.DialogInterface dialog, int which) {
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         sLogger.onCreateView();
+        //null
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/base/BaseDialogFragment.java]: What?

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/GGMapGestureListener.java [hunk 9]
```diff
@@ -11,7 +11,8 @@
 import com.teamagam.gimelgimel.app.view.viewer.gestures.SimpleOnMapGestureListener;
 
 public class GGMapGestureListener extends SimpleOnMapGestureListener
-        implements OnMapGestureListener {
+        implements OnMapGestureListener
+        {
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/GGMapGestureListener.java]: Java conventions - move the bracket to the same line.

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/data/symbols/EntityMessageSymbolizer.java [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * message Symbolizer that uses visitor pattern.
+ */
+public class EntityMessageSymbolizer implements
+        ViewerFragment.MessageSymbolizer {
+
+    Symbol mSymbolResult;
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/data/symbols/EntityMessageSymbolizer.java]: Have you forgot the accessor?

File: app/src/main/java/com/teamagam/gimelgimel/app/model/ViewsModels/MessageMapEntitiesViewModel.java (hunk 3 above)
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/model/ViewsModels/MessageMapEntitiesViewModel.java:46]: !resolve

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java (hunk 4 above)
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java]: !fix

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendGeographicMessageDialog.java (hunk 7 above)
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendGeographicMessageDialog.java]: !fix

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendGeographicMessageDialog.java (hunk 6 above)
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendGeographicMessageDialog.java]: !fix

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendGeographicMessageDialog.java (hunk 5 above)
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendGeographicMessageDialog.java]: !resolve

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/GGMapGestureListener.java (hunk 9 above)
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/GGMapGestureListener.java]: !fix

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/base/BaseDialogFragment.java (hunk 8 above)
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/base/BaseDialogFragment.java]: !resolve

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/data/symbols/EntityMessageSymbolizer.java (hunk 10 above)
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/data/symbols/EntityMessageSymbolizer.java]: !fix

File: app/src/main/java/com/teamagam/gimelgimel/app/model/ViewsModels/MessageMapEntitiesViewModel.java (hunk 3 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/model/ViewsModels/MessageMapEntitiesViewModel.java:46]: דברנו על זה. !resoved

Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/model/ViewsModels/MessageMapEntitiesViewModel.java:46]: !resolved

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java (hunk 4 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java]: !fixed

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendGeographicMessageDialog.java (hunk 5 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendGeographicMessageDialog.java]: the ArrayAdapter.createFromResource returns CharSequence. I have nothing to do about it.

Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendGeographicMessageDialog.java]: !completed

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendGeographicMessageDialog.java (hunk 6 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendGeographicMessageDialog.java]: !fixed

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendGeographicMessageDialog.java (hunk 7 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendGeographicMessageDialog.java]: !fixed

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/base/BaseDialogFragment.java (hunk 8 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/base/BaseDialogFragment.java]: !fixed

Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/base/BaseDialogFragment.java]: the next method returns null. but for android lifecycle we must call it.

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/GGMapGestureListener.java (hunk 9 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/GGMapGestureListener.java]: !fixed

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/data/symbols/EntityMessageSymbolizer.java (hunk 10 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/data/symbols/EntityMessageSymbolizer.java]: !fixed

File: GimelGimel.iml (hunk 1 above)
Comment (teamagam1) [inline GimelGimel.iml]: !!fixed

**Pull request merged**